### PR TITLE
Make RTD not the main check for sphinx warnings

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
`sphinx-build: error: argument -n: ignored explicit argument '4'` Need to remove the xdist bit.

**Comment by nabobalis** [comment]
Still got:
```
sphinx-build: error: argument -n: ignored explicit argument '4'
ERROR: InvocationError for command /home/vsts/work/1/s/.tox/build_docs/bin/sphinx-build --color -W --keep-going -b html -d _build/.doctrees . _build/html --junitxml=junit/test-results.xml --cov-report=xml:/home/vsts/work/1/s/coverage.xml -n=4 (exited with code 2)
build_docs: Shutting down tox-pypi-filter server
```
Not sure why.

**Comment by nabobalis** [comment]
RTD just says:
```
Error
There was a problem with Read the Docs while building your documentation. Please try again later. However, if this problem persists, please report this to us with your build id (11796097).

Report any build issues here.
```

**Comment by nabobalis** [comment]
```
usage: sphinx-build [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]
sphinx-build: error: unrecognized arguments: --junitxml=junit/test-results.xml --cov-report=xml:/home/vsts/work/1/s/coverage.xml
ERROR:
```

**Comment by nabobalis** [comment]
Don't we build the docs on glue-solar without this issue?

**Comment by nabobalis** [comment]
Oh yeah just remove:
https://github.com/sunpy/sunpy/blob/master/tox.ini#L93

File: azure-pipelines.yml [hunk 1]
```diff
@@ -83,6 +83,10 @@ jobs:
 
       - linux: py38-devdeps
 
+      - linux: build_docs
+        posargs: " "
+        coverage: false
```
Comment (nabobalis) [inline azure-pipelines.yml:88]: If you remove the posargs from the tox build we can remove these two lines.

**Comment by Cadair** [comment]
posargs is useful for things like `-aE` to clean build the docs though (which is in our  dev docs)

**Comment by nabobalis** [comment]
Do as you want.

**Comment by nabobalis** [comment]
Looks like it works now.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -102,5 +107,5 @@ jobs:
         - py38_test
         - py38_test_online
         - py38_test_32bit
-        - py38_test_oldestdeps
+        - py37_test_oldestdeps
```
Comment (wtbarnes) [inline azure-pipelines.yml:110]: What's the reason for moving this to 3.7?

Comment (nabobalis) [reply azure-pipelines.yml:110]: My reasoning but cadair might have a different one. 

Oldest deps should be tested on the oldest python version we support.

Comment (Cadair) [reply azure-pipelines.yml:110]: I didn't actually change the version, just needed to update the wheels depends list because someone forgot it.

**Pull request merged**